tree_unique_args/src/ast.rs

```diff
@@ -1,5 +1,26 @@
 use crate::{Expr, Expr::*, Id, Order};
 
+impl Expr {
+    pub fn eq_ignoring_ids(&self, other: &Expr) -> bool {
+        let mut copy = other.clone();
+        give_fresh_ids(&mut copy);
+        self == &copy
```

Comment:
dumb q; why does this work?

Reply:
Not a dumb question- I added a comment
```
    /// Check that two expressions are the same ignoring their ids.
    /// To do this, simply assign them all new ids.
    /// If they are the same expression, they will get the same ids
    /// since `give_fresh_ids` is deterministic.
```